include/crpropa/Common.h

```diff
@@ -67,7 +67,21 @@ inline double pow_integer<0>(double base)
   return 1;
 }
 
-
+// - input:  function over which to integrate, integration limits A and B
+// - output: 8-points Gauß-Legendre integral
+template<typename Integrand>
+inline double gaussInt(Integrand&& integrand, double A, double B) {
+	static const double X[8] = {.0950125098, .2816035507, .4580167776, .6178762444, .7554044083, .8656312023, .9445750230, .9894009349};
```

Comment:
These integration constants should not be static inside the template but a static constant outside. Otherwise there will be a new static array every time the template is evaluated.